### DISCUSSION TITLE
Jaeger to Open-telemetry means?

### DISCUSSION THREAD

**Discussion by SamYuan1990** [open-telemetry/opentelemetry-go#2742] [General]
Hello everyone,

I want to migrate from Jaeger to Open-telemetry.
So far on my k8s I have prometheus and jaeger operators used in my project at infrastructures level as side car.
and I am done with sample for use jaeger sdk in golang language.

I want to know, what's migration from Jaeger to Open-telemetry mean to me?
- [ ] 1. do I need to replace jaeger operator with open-telemetry operator at infrastructure level?
- [ ] 2. is there any UI as jaeger for open-telemetry?
- [ ] 3. I need to search span base on specific value(tag `key=value`) if there is a UI, do we have any supporting?
- [ ] 4. do I just need to replace golang implemented in code by open-telemetry by jaeger?
- [ ] 5. any configuration upgrade needed?

Any Others unlisted items?

ref to https://opentelemetry.io/docs/migration/opentracing/#migration-overview
I suppose item 1, 4 are needed. Not sure about 2/3/5

But, well...
```
Progressively re-instrument your application using the OpenTelemetry API. There is no hard requirement to remove existing OpenTracing API calls from your application, they will continue to work.
```

Thanks and Regards
Sam

**Comment by dmathieu** [comment]
Hi,

OpenTelemetry is not a full replacement for jaeger. It is a vendor-agnostic set of libraries and component meant to perform telemetry.
This means your application will be able to emit traces, metrics and logs in a vendor agnostic way, and send them to any (compatible) vendor. Jaeger can be a vendor.

> 1. do I need to replace jaeger operator with open-telemetry operator at infrastructure level?

If your tracing UI remaing jaeger, you will still need the operator, yes. You may also need the otel collector to emit all data in a single place. But that's not mandatory.

> 2. is there any UI as jaeger for open-telemetry?

No. See my explanation above.

 > 4. I need to search span base on specific value(tag key=value) if there is a UI, do we have any supporting?

See above again.

> 5. do I just need to replace golang implemented in code by open-telemetry by jaeger?

Yes. Replacing the libraries you call within your code would be the main part of your migration.

> 6. any configuration upgrade needed?

This is quite broad. Yes, as there are some things that won't work exactly the same way between otel and jaeger. They are entirely different components, so you will need to rethink how your configuration is done.

**Reply by SamYuan1990** [reply]
I see, it seems lots of work to do, and @dmathieu , could you please find me some sample or this there any plan for otel fully replace jaeger or there any things as CNI spec/OCI spec in coming? so that as a user I am able to switch between otel and jaeger in quick.

**Reply by dmathieu** [reply]
Otel has no plan to provide a UI to query data. So no, there's no plan to fully replace jaeger.
